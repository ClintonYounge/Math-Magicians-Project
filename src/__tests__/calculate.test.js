import calculate from '../logic/calculate';

describe('calculate', () => {
  it('should return an object with total: 0, next: null, and operation: null when buttonName is AC', () => {
    const result = calculate({ total: 100, next: 50, operation: '+' }, 'AC');
    expect(result).toMatchSnapshot({ total: null, next: null, operation: null }, `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`);
  });

  it('should return an object with updated next value when buttonName is a number', () => {
    const result = calculate({ total: 100, next: 50, operation: '+' }, '2');
    expect(result).toMatchSnapshot({ total: 100, next: '502', operation: '+' }, `
Object {
  "next": "502",
  "operation": "+",
  "total": 100,
}
`);
  });

  it('should return an object with updated next value when buttonName is .', () => {
    const result = calculate({ total: 100, next: '50', operation: '+' }, '.');
    expect(result).toMatchSnapshot({ total: 100, next: '50.', operation: '+' }, `
Object {
  "next": "50.",
  "operation": "+",
  "total": 100,
}
`);
  });

  it('should return an object with updated next value when buttonName is +/-', () => {
    const result = calculate({ total: 100, next: '50', operation: '+' }, '+/-');
    expect(result).toMatchSnapshot({ total: 100, next: '-50', operation: '+' }, `
Object {
  "next": "-50",
  "operation": "+",
  "total": 100,
}
`);
  });
});
