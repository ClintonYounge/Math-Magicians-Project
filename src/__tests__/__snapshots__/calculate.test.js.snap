// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate should return an object with total: 0, next: null, and operation: null when buttonName is AC: 
Object {
  "next": null,
  "operation": null,
  "total": null,
}
 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;

exports[`calculate should return an object with updated next value when buttonName is +/-: 
Object {
  "next": "-50",
  "operation": "+",
  "total": 100,
}
 1`] = `
Object {
  "next": "-50",
  "operation": "+",
  "total": 100,
}
`;

exports[`calculate should return an object with updated next value when buttonName is .: 
Object {
  "next": "50.",
  "operation": "+",
  "total": 100,
}
 1`] = `
Object {
  "next": "50.",
  "operation": "+",
  "total": 100,
}
`;

exports[`calculate should return an object with updated next value when buttonName is a number: 
Object {
  "next": "502",
  "operation": "+",
  "total": 100,
}
 1`] = `
Object {
  "next": "502",
  "operation": "+",
  "total": 100,
}
`;
